php/commands/package.php

```diff
@@ -183,6 +183,13 @@ public function install( $args, $assoc_args ) {
 		$json_manipulator->addMainKey( 'name', 'wp-cli/wp-cli' );
 		$json_manipulator->addLink( 'require', $package_name, $version );
 		$json_manipulator->addConfigSetting( 'secure-http', true );
+
+		// If the composer file does not contain the current package index url, refresh the repository definition.
+		if ( false === strpos( $composer_backup, self::PACKAGE_INDEX_URL ) ) {
```

Comment:
This is a bit too blunt.

We should specifically check the repository definition, and only replace `http://wp-cli.org/package-index/`